sys/kern/device.c

```diff
@@ -91,3 +91,14 @@ resource_t *device_take_resource(device_t *dev, res_type_t type, int rid,
 
   return rle->res;
 }
+
+device_t *device_if_find_impl(device_t *dev, size_t iface,
+                              size_t method_offset) {
+  while ((!dev->parent->driver->interfaces[iface]) ||
```

Comment:
Maybe the following would be more readable
```C
static bool device_if_implemented(void *interface, size_t offset) {
  return *(void **)(interface + offset) != NULL;
}

...
for (;;) {
  device_t *parent = dev->parent;
  assert(parent);
  void *interface = parent->interfaces[iface];
  if (interface && device_if_implemented(interface, method_offset))
    return dev;
  dev = parent;
}
return NULL;
```

Reply:
Idk. I'll think about it. It looks like a sligh overkill to make a separate function for that, but maaybe that makes it more readable. The name is a bit misleading tho. Also, this should be inline. How about a comment explaining the check?

Reply:
Done!